wdl-grammar/src/v1/lint/header_comment.rs

```diff
@@ -0,0 +1,178 @@
+//! Header comments are full line comments starting with a double pound sign
+
+use std::collections::VecDeque;
+
+use nonempty::NonEmpty;
+use pest::iterators::Pair;
+use wdl_core::concern::code;
+use wdl_core::concern::lint;
+use wdl_core::concern::lint::Group;
+use wdl_core::concern::lint::Rule;
+use wdl_core::concern::Code;
+use wdl_core::file::Location;
+use wdl_core::Version;
+
+use crate::v1;
+
+/// Detects header comments declaration
+#[derive(Debug)]
+pub struct HeaderComment;
+
+impl<'a> HeaderComment {
+    /// Creates a warning for header comments without double pound sign
```

Comment:
@adthrasher do we want this part of the warning? When I wrote the rule I didn't intend to include this case, but maybe it should be warned on.

Historically we've had it both ways; and I'm fairly certain every time we had a case of mixing single-pound and double-pound header comments we've eventually either deleted the single ones or "upgraded" them to double-pound comments.

For additional context: double-pound header comments are meant to be copied into `sprocket` (or `wdldoc`) built docs, and single-pound "header" comments are just regular comments meant to be ignored by tooling.

A potential convention that keeps this bit of code would be:
```
## sprocket comment

version 1.1
# "regular" comment sprocket will ignore
# Copyright St. Jude Children's Research Hospital

import "../foo.wdl"
```
We've dubbed the header+version declaration the "document preamble", I think it makes sense to keep the preamble well defined (**only** header comments and the version).